// kinopoisk-api/lib/kinopoisk-api.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getFilm = getFilm;
exports.getStaff = getStaff;
exports.getGallery = getGallery;
exports.getSimilar = getSimilar;
exports.getReviews = getReviews;
exports.getSeance = getSeance;
exports.searchFilms = searchFilms;

var _nodeFetch = require('node-fetch');

var _nodeFetch2 = _interopRequireDefault(_nodeFetch);

var _lodash = require('lodash.flatten');

var _lodash2 = _interopRequireDefault(_lodash);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var api = 'http://api.kinopoisk.cf';
var f = function f(uri) {
  return (0, _nodeFetch2.default)(api + '/' + uri).then(function (res) {
    return res.json();
  });
};

function getFilm(filmID) {
  return f('getFilm?filmID=' + filmID);
}

function getStaff(filmID) {
  return f('getStaff?filmID=' + filmID);
}

function getGallery(filmID) {
  return f('getGallery?filmID=' + filmID);
}

function getSimilar(filmID) {
  return f('getSimilar?filmID=' + filmID);
}

function getReviews(filmID) {
  return f('getReviews?filmID=' + filmID);
}

function getSeance(filmID) {
  return f('getSeance?filmID=' + filmID);
}

function searchFilms() {
  for (var _len = arguments.length, keywords = Array(_len), _key = 0; _key < _len; _key++) {
    keywords[_key] = arguments[_key];
  }

  return f('searchFilms?keyword=' + (0, _lodash2.default)(keywords).join(','));
}
